File: int32 to IPv4/uint32_to_ip.py
#!/usr/bin/env python

# My solution
def ip(n):
    return '.'.join(str((n := n % 256**(4-i)) // 256**(3-i)) for i in range(4))


# My favorite solution
def int32_to_ip(int32):
    return "{}.{}.{}.{}".format(*int32.to_bytes(4, 'big'))

File: int32 to IPv4/uint32_to_ip.rs

// My solution
fn ip(n: u32) -> String {
    let ar = n.to_be_bytes();
    return format!("{}.{}.{}.{}", ar[0],ar[1],ar[2],ar[3]);
}


// My favorite solution
use itertools::Itertools

fn int32_to_ip(n: u32) -> String {
    n.to_be_bytes().iter().join(".")
}
